stories/CodeView.stories.tsx

```diff
@@ -34,30 +38,129 @@ const CSS = `html, body {
   display: none;
 }`;
 
-storiesOf('CodeView', module).addWithChapters('all variants', {
-  chapters: [
-    {
-      sections: [
-        {
-          title: 'unsupported mime type',
-          sectionFn: () =>
-            renderWithStoreAndRouter(<CodeView mimeType="" content={JS} />),
-        },
-        {
-          title: 'application/javascript',
-          sectionFn: () =>
-            renderWithStoreAndRouter(
-              <CodeView mimeType="application/javascript" content={JS} />,
-            ),
-        },
-        {
-          title: 'text/css',
-          sectionFn: () =>
-            renderWithStoreAndRouter(
-              <CodeView mimeType="text/css" content={CSS} />,
-            ),
-        },
-      ],
-    },
-  ],
-});
+const render = (moreProps = {}) => {
+  const props = {
+    content: JS,
+    linterMessagesByLine: undefined,
+    mimeType: 'application/javascript',
+    ...moreProps,
+  };
+  return renderWithStoreAndRouter(<CodeView {...props} />);
+};
+
+const renderJSWithMessages = (
+  messages: Partial<LinterMessage>[],
+  moreProps: Partial<CodeViewProps> = {},
+) => {
+  const stubFile = 'lib/some-file.js';
+  const map = getMessageMap(
+    createFakeExternalLinterResult({
+      messages: messages.map((msg) => {
+        return {
+          ...msg,
+          file: stubFile,
+        };
+      }),
+    }),
+  );
+
+  return render({
+    content: JS,
+    mimeType: 'application/javascript',
+    linterMessagesByLine: map[stubFile].byLine,
+    ...moreProps,
+  });
+};
+
+storiesOf('CodeView', module)
+  .addWithChapters('mime types', {
+    chapters: [
+      {
+        sections: [
+          {
+            title: 'unsupported mime type',
+            sectionFn: () => render({ mimeType: '', content: JS }),
+          },
+          {
+            title: 'application/javascript',
+            sectionFn: () =>
+              render({ mimeType: 'application/javascript', content: JS }),
+          },
+          {
+            title: 'text/css',
+            sectionFn: () => render({ mimeType: 'text/css', content: CSS }),
+          },
+        ],
+      },
+    ],
+  })
+  .addWithChapters('linter messages', {
```

Comment:
That's super cool! Why did you split the stories into two chapters though? I thought we wanted to have everything on a single page so that we see when UI things break? (this will be important for mvp-polish). I am fine with two chapters but we should know when to use 2 chapters.

Reply:
The initial reason for why I wanted stories on one page was to empower this kind of development flow:

1. Make a change to the code
2. Let storybook auto-reload
3. View a single storybook page to check if you broke anything

This doesn't really fit with the `CodeView` stories because the mime-type stories take up so much of the screen. You already have to scroll to see the stories so adding more to the bottom doesn't seem useful. It would be easier to click a new chapter link.

So, yeah, I think a rule of thumb would be to ask, do these screens all roughly fit on the page in full screen mode? If the answer is no then it makes sense to break them into new chapters.

Reply:
That works for me, thanks!